app/models/application_record.rb

```diff
@@ -88,15 +88,18 @@ def self.eeo_data(start_date, end_date, department_ids)
       [option, specific_records.count, specific_records.interview_count]
     end
     records[:genders] = gender_eeo_data(start_date, end_date, department_ids)
-    records[:male_ethnicities] = all_ethnicities.map do |ethnicity|
-      specific_records = ethnicity_records
-                         .where ethnicity: ethnicity, gender: 'Male'
-      [ethnicity, specific_records.count, specific_records.interview_count]
-    end
-    records[:female_ethnicities] = all_ethnicities.map do |ethnicity|
-      specific_records = ethnicity_records
-                         .where ethnicity: ethnicity, gender: 'Female'
-      [ethnicity, specific_records.count, specific_records.interview_count]
+    all_ethnicities.map do |ethnicity|
+      combined_records = ethnicity_records.with_gender
+                                          .where(ethnicity: ethnicity)
+      male_records = combined_records.where ethnicity: ethnicity, gender: 'Male'
+      records[:male_ethnicities] << [ethnicity,
```

Comment:
So, this is still the same code repeated twice with just 'Male' and 'Female' being different. Let's work together on this tomorrow.